plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/LambdaLineMarker.kt

```diff
@@ -79,7 +79,7 @@ class LambdaLineMarker : LineMarkerProviderDescriptor() {
 
     private fun shouldShowLineMarker(psiFile: PsiFile, handler: String, runtimeGroup: RuntimeGroup): Boolean {
         val project = psiFile.project
-        return LambdaSettings.getInstance(project).showAllHandlerGutterIcons ||
+        return LambdaSettings.getInstance().showAllHandlerGutterIcons ||
```

Comment:
why did we convert this to an application setting?

Reply:
Or I guess there's another way is getting the project from some data context, but it does have the risk of having a nullable project, I think.

Reply:
lets just keep as a separate child

Reply:
sure